### PR TITLE
Feature/speech prediction

### PR DESCRIPTION
**Pull request by JakeBrawer**
Created  web interface for human collaboration speech prediction project.

### CODE REVIEW

File: human_robot_collaboration/launch/flatpack_furniture.launch [hunk 1]
```diff
@@ -36,7 +36,21 @@
...
+    </node>
+
+
+    <rosparam param = "hsv_detector/objects_db">
+        "screwdriver" : { size: [0.099, 0.030], HSV: { H: [160,  10], S: [ 70, 166], V: [10,  66]}}
+         <!-- This should be the blue box -->
+        "screws_box"  : { size: [0.111, 0.111], HSV: { H: [ 80, 130], S: [120, 216], V: [70, 146]}}
+         <!-- This should be the green box -->
+        "brackets_box": { size: [0.111, 0.111], HSV: { H: [ 30, 100], S: [ 80, 246], V: [20,  96]}}
+    </rosparam>
```
Comment (alecive) [inline human_robot_collaboration/launch/flatpack_furniture.launch]: Why did you have to modify this file? Please revert.

File: human_robot_collaboration/launch/speech_prediction.launch [hunk 2]
```diff
@@ -1,67 +0,0 @@
-<launch>
```
Comment (alecive) [inline human_robot_collaboration/launch/speech_prediction.launch]: So this file is not needed any more?

File: human_robot_collaboration/web-interface/speech_predicition/interface.html [hunk 3]
```diff
@@ -1,4 +1,4 @@
-<!DOCTYPE html>
+ <!DOCTYPE html>
```
Comment (alecive) [inline human_robot_collaboration/web-interface/speech_predicition/interface.html]: added useless space

File: human_robot_collaboration/web-interface/speech_predicition/interface.html [hunk 4]
```diff
@@ -31,12 +31,38 @@
...
+                    <text></text>
+            </div>
+        </div>
+    </div>
+        <div class="row">
             <div class="col-xl-6 text-center" id="left_table_div">
+                    <h5 align="center">Left Table</h5>
                 <div id="left-svg-container" class="svg-container">
-                    <svg></svg>
+                    <svg> </svg>
```
Comment (alecive) [inline human_robot_collaboration/web-interface/speech_predicition/interface.html]: useless space?

File: human_robot_collaboration/web-interface/speech_predicition/interface.html [hunk 5]
```diff
@@ -148,6 +175,7 @@
        document.getElementById('container-default').style.display = 'block';
        document.getElementById('container-get_pass').style.display = 'none';
        document.getElementById('container-cleanup').style.display = 'none';
+        aC
```
Comment (alecive) [inline human_robot_collaboration/web-interface/speech_predicition/interface.html]: this is definitely not needed.

File: human_robot_collaboration/web-interface/speech_predicition/ros.js [hunk 6]
```diff
@@ -71,20 +71,45 @@ var errorPressedR = new ROSLIB.Topic({
...
+    name: '/action_provider/left/state',
+    messageType : 'human_robot_collaboration_msgs/ArmState'
+});
+
 // Service Client to interface with the left arm
 var leftArmService  = new ROSLIB.Service({
   ros : ros,
   name: '/action_provider/service_left',
-  messageType : 'human_robot_collaboration_msgs/DoAction'
+  messageType : 'baxter_collaboration_msgs/DoAction'
```
Comment (alecive) [inline human_robot_collaboration/web-interface/speech_predicition/ros.js]: revert back to `human_robot_collaboration_msgs`

File: human_robot_collaboration/web-interface/speech_predicition/ros.js [hunk 7]
```diff
@@ -71,20 +71,45 @@ var errorPressedR = new ROSLIB.Topic({
...
 });
 
 // Service Client to interface with the right arm
 var rightArmService = new ROSLIB.Service({
   ros : ros,
   name: '/action_provider/service_right',
-  messageType : 'human_robot_collaboration_msgs/DoAction'
+  messageType : 'baxter_collaboration_msgs/DoAction'
```
Comment (alecive) [inline human_robot_collaboration/web-interface/speech_predicition/ros.js]: revert back to `human_robot_collaboration_msgs`

File: human_robot_collaboration/web-interface/speech_predicition/ros.js [hunk 8]
```diff
@@ -71,20 +71,45 @@ var errorPressedR = new ROSLIB.Topic({
...
 });
 
+
```
Comment (alecive) [inline human_robot_collaboration/web-interface/speech_predicition/ros.js]: useless space

File: human_robot_collaboration/web-interface/speech_predicition/style.css [hunk 9]
```diff
@@ -24,7 +24,7 @@ Class prefix        .col-xs-        .col-sm-    .col-md-    .col-lg-    .col-xl-
 html {
   font-size: 18px;
 }
-
+t
```
Comment (alecive) [inline human_robot_collaboration/web-interface/speech_predicition/style.css]: not needed.

File: human_robot_collaboration/web-interface/speech_predicition/tables.js [hunk 10]
```diff
@@ -1,51 +1,122 @@
...
+
+function armInfoCallback(msg,arm){
+
+    console.log("Receiving arm info!: " + msg.state);
+    var s = d3.select("#" + arm + "_baxter_state");
+    var a = d3.select("#" + arm + "_baxter_action");
+    var o = d3.select("#" + arm + "_baxter_object");
+
+    s.select("text")
+        .text("STATE: " + msg.state);
```
Comment (alecive) [inline human_robot_collaboration/web-interface/speech_predicition/tables.js]: alignment

File: human_robot_collaboration/web-interface/speech_predicition/tables.js [hunk 11]
```diff
@@ -1,51 +1,122 @@
...
+
+    a.select("text")
+        .text("ACTION: " + msg.action);
```
Comment (alecive) [inline human_robot_collaboration/web-interface/speech_predicition/tables.js]: alignment

File: human_robot_collaboration/web-interface/speech_predicition/tables.js [hunk 12]
```diff
@@ -1,51 +1,122 @@
...
+
+    o.select("text")
+        .text("OBJECT: " + msg.object);
```
Comment (alecive) [inline human_robot_collaboration/web-interface/speech_predicition/tables.js]: alignment

File: human_robot_collaboration_lib/src/robot_perception/cartesian_estimator.cpp [hunk 13]
```diff
@@ -188,6 +188,26 @@ bool CartesianEstimator::publishObjects()
             object_cnt.pose = objs[i]->pose;
             object_cnt.id   = i;
             object_cnt.name = objs[i]->getName();
+
+
```
Comment (alecive) [inline human_robot_collaboration_lib/src/robot_perception/cartesian_estimator.cpp]: useless space

File: human_robot_collaboration_lib/src/robot_perception/cartesian_estimator.cpp [hunk 14]
```diff
@@ -188,6 +188,26 @@ bool CartesianEstimator::publishObjects()
...
+            geometry_msgs::Point cent;
```
Comment (alecive) [inline human_robot_collaboration_lib/src/robot_perception/cartesian_estimator.cpp]: indentation

**Review by alecive — CHANGES_REQUESTED** [review]
The main thing you should change is filenames (to match the containing folder).

File: human_robot_collaboration/launch/flatpack_furniture.launch (hunk 1 above)
Comment (JakeBrawer) [reply human_robot_collaboration/launch/flatpack_furniture.launch]: I didnt mean too, sorry

File: human_robot_collaboration/web-interface/speech_predicition/interface.html (hunk 5 above)
Comment (JakeBrawer) [reply human_robot_collaboration/web-interface/speech_predicition/interface.html]: This is the command for opening up slack in emacs :^)

File: human_robot_collaboration/web-interface/speech_predicition/ros.js [hunk 15]
```diff
@@ -191,8 +227,32 @@ function callback(e) {
...
+            console.log("Requested rosbag recording start!");
+            startExperiment.callService(req,function(res){
+                console.log("Got response!");
+            });
+        }
+
+        else if (obj.includes("STOP"))
+        {
+            var req = new ROSLIB.ServiceRequest();
+            var res = new ROSLIB.ServiceResponse();
```
[the remark this answers is not in the document]
Comment (alecive) [reply human_robot_collaboration/web-interface/speech_predicition/ros.js]: @houndci-bot has a point. We should define `req` and `res` once in the beginning, and reuse them later on.

File: human_robot_collaboration/launch/flatpack_furniture.launch [hunk 16]
```diff
@@ -36,7 +36,6 @@
     </rosparam>
 
     <node pkg="human_robot_collaboration" type="baxter_display"  name="baxter_display"  output="screen"/>
-    <node pkg="human_robot_collaboration" type="flatpack_action_provider" name="flatpack_action_provider" output="screen"/>
```
Comment (alecive) [inline human_robot_collaboration/launch/flatpack_furniture.launch]: You still have this to be taken care of.

File: human_robot_collaboration/launch/speech_prediction.launch [hunk 17]
```diff
@@ -1,67 +1,104 @@
...
+        "leg_1"         : 150
+        "leg_2"         : 151
+        "leg_3"         : 152
+        "leg_4"         : 153
     </rosparam>
 
     <!-- Objects database for the right arm -->
     <rosparam param = "action_provider/objects_right">
-        "screwdriver"  : 1
+        "screwdriver"    : 1
```
Comment (alecive) [inline human_robot_collaboration/launch/speech_prediction.launch]: indentation

**Review by alecive — CHANGES_REQUESTED** [review]

**Review by alecive — APPROVED** [review]

**Pull request merged**